apps/storybook-e2e/src/e2e/components/rich-data-table/vl-rich-data-table.stories.cy.ts

```diff
@@ -1,14 +1,267 @@
+};
+
+describe('story vl-rich-data-table default', () => {
+    it('should set data in the table', () => {
+        cy.visit(richDataTableUrl);
+
+        const { data: dataRows } = {
+            data: [
+                { id: 0, name: 'Project #1', owner: 'Jan Jansens' },
+                { id: 1, name: 'Project #2', owner: 'Marie Vermeersch' },
+            ],
+        };
+
+        shouldMatchTableData(dataRows);
+    });
+});
+
+const sortingData = {
+    data: [
+        { id: 0, name: 'Water', owner: 'Kevin Jansens' },
+        { id: 1, name: 'Vuur', owner: 'Anton Vanherrewege' },
+        { id: 2, name: 'Aarde', owner: 'Hedwig Jansens' },
+    ],
+};
+
+describe('story vl-rich-data-table - sorting', () => {
+    before(() => cy.visit(`${richDataTableSortingUrl}`));
+
+    const { data: rowData } = {
+        data: [
+            { id: 0, name: 'Water', owner: 'Kevin Jansens' },
+            { id: 1, name: 'Vuur', owner: 'Anton Vanherrewege' },
+            { id: 2, name: 'Aarde', owner: 'Hedwig Jansens' },
+        ],
+    };
+
+    it('should set data in the table', () => {
+        shouldMatchTableData(rowData);
+    });
+
+    it('should have default sorting as ascending', () => {
+        const keyToSortOn = 'id';
+        const direction = 'asc';
 
-describe('story vl-rich-data-table', () => {
-    it('as a user, I can see the rich-data-table', () => {
-        cy.visit(`${richDataTableUrl}`);
-        cy.get('vl-rich-data-table')
-            .shadow()
-            .find('tr')
-            .then((rows) => {
-                expect(rows.length).to.equal(3);
-            });
+        shouldHaveActiveSorterAndMatchExpectedDirection(keyToSortOn, direction);
+
+        executeForEveryRow((row, rowIndex) =>
+            shouldSortFieldsAsExpected(row, rowIndex, rowData, keyToSortOn, direction)
+        );
+    });
+
+    it('should have sorting as descending when active sorter is clicked once', () => {
+        const keyToSortOn = 'id';
+        const direction = 'desc';
+
+        clickSorterForField(keyToSortOn);
+
+        shouldHaveActiveSorterAndMatchExpectedDirection(keyToSortOn, direction);
+
+        executeForEveryRow((row, rowIndex) =>
+            shouldSortFieldsAsExpected(row, rowIndex, rowData, keyToSortOn, direction)
+        );
+    });
+
+    it('should have no sorting, when sorter is currently sorting descending', () => {
+        const keyToSortOn = 'id';
+        clickSorterForField(keyToSortOn);
+
+        shouldHaveSorterAndDirectionShouldBeHidden(keyToSortOn);
+    });
+});
+
+describe('story vl-rich-data-table - filter', () => {
+    before(() => cy.visit(`${richDataTableFilterUrl}`));
+    const data = rowsForFiltering;
+    const filterField = 'name';
+    const filterValue = 'Grond';
+
+    it('should be able to find data', () => {
+        shouldMatchTableData(data);
+    });
+
+    it('should be able to filter on a field', () => {
+        filterOnGivenField(data, filterField, filterValue);
```

Comment:
filterOnGivenField wordt enkel hier gebruikt, kan je hier de code zetten in plaats van er een methode rond te wrappen?

Reply:
done